.github/workflows/publish.yml

```diff
@@ -0,0 +1,12 @@
+name: Publish to pub.dev
+
+on:
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+
+jobs:
+  publish:
+    permissions:
+      id-token: write
+    uses: dart-lang/setup-dart/.github/workflows/publish.yml@v1
```

Comment:
You want to use our template here https://github.com/famedly/matrix-dart-sdk/blob/main/.github/workflows/publish.yml

also please setup the `pub.dev` env in github settings